src/2ls/preprocessing_util.cpp

```diff
@@ -112,13 +115,22 @@ bool twols_parse_optionst::unwind_goto_into_loop(
     loopst loops;
     Forall_goto_program_instructions(i_it, body)
     {
-      if(i_it->is_backwards_goto())
```

Comment:
(independent of this PR) If you think that this is a bug in goto program it may be worth creating a unit test for goto program that reproduces the issue.

Reply:
I think it may be a bug, though I am not completely sure. In one of our regression tests (`termination/ite4`), this call resulted in a segfault. We tried to investigate it but it was very weird - we successfully entered the method (hence `i_it` is fine) but the number of targets of the goto was huge so I'd guess that's where the bug came from (iterating over the targets and accessing a weird memory location).

It may have already been fixed in a later version of CBMC. We could try to revert this commit once we are close to CBMC:develop and if it still fails, we can try to find the root cause + add a unit test. Or maybe could you try it if you have latest CBMC setup? The reproducer is very simple:

```

int foo(int x)
{
  int res;
  if(x) return 0;
  else return 1;
}

int bar(int x)
{
  int res;
  if(x) res = 0;
  else res = 1;
  return res;
}

void main()
{
  int x;
  int y = foo(x);
  assert(0<=y && y<=1);
  y = bar(x);
  assert(0<=y && y<=1);
}
```
If I remember correctly, the segfault happened right in the first function at `if`.